projects/plugins/social/src/class-rest-social-note-controller.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * The Social Note Controller class.
+ * Registers the REST routes for Social.
+ *
+ * @package automattic/jetpack-social-plugin
+ */
+
+namespace Automattic\Jetpack\Social;
+
+use Automattic\Jetpack\Connection\Rest_Authentication;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Server;
+
+/**
+ * Registers the REST routes for Social.
+ */
+class REST_Social_Note_Controller extends WP_REST_Controller {
+	const SOCIAL_SHARES_POST_META_KEY = '_publicize_shares';
+	/**
+	 * Registers the REST routes for Social.
+	 *
+	 * @access public
+	 * @static
+	 */
+	public function register_rest_routes() {
+		register_rest_route(
+			'jetpack/v4',
+			'/social/update-post-meta',
```

Comment:
I wonder if we shouldn't be so specific about the implementation of this, and the endpoint should be something like /social/shares the post ID could be part of the route to get the details for that post. I think we could also use it to get a summary of shares information for the site.

Alternatively, we could use the core meta endpoints for posts and [register_rest_field ](https://developer.wordpress.org/reference/functions/register_rest_field/) (I notice Manzoor has left a comment there about being able to add validation to the schema) in order to register the meta value. The problem with that is that I'm not sure how we would handle the authentication, but my guess is that we could hook into the request with a filter somehow.

Reply:
>I wonder if we shouldn't be so specific about the implementation of this, and the endpoint should be something like /social/shares the post ID could be part of the route to get the details for that post. I think we could also use it to get a summary of shares information for the site.

I have made the implementation generic by renaming the name to /social/shares/posts/:ID. The route handler checks if the post meta is set, but we can extend the route to update anything else share information related in the future.